src/suit/js/actions.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import {get, set, cloneDeep} from 'lodash';
+import {ActiveTableButton} from 'firefly/tables/ui/ActiveTableButton.jsx';
+import {getTblById, getCellValue, makeTblRequest} from 'firefly/tables/TableUtil.js';
+import {getViewer} from 'firefly/api/ApiViewer.js';
+import {LC} from 'firefly/templates/lightcurve/LcManager.js';
+
+
+export function timeSeriesButton() {
+    return (
+        <div key='launchTimeSeries'>
+            <ActiveTableButton
+                title = 'Launch Time Series Viewer'
+                onChanges = {checkForTimeSeriesData}
+                onClick = {launchTimeSeries}
+            />
+        </div>
+    );
+}
+
+function launchTimeSeries({tbl_id, highlightedRow}) {
+    const tableModel = getTblById(tbl_id) || {};
+    if (isTimeSeriesTable(tableModel)) {
+        const req = cloneDeep(get(tableModel, 'request'));
+        req.tbl_id = LC.RAW_TABLE;
+        set(req, 'META_INFO.tbl_id', LC.RAW_TABLE);
+        req && getViewer(undefined, 'timeseries').showTable(req);
+    } else {
+        const objectId = getCellValue(getTblById(tbl_id), highlightedRow, 'objectId');
+        const META_INFO = {[LC.META_MISSION]: 'lsst_sdss'};
+        const req = makeTblRequest('LSSTLightCurveQuery', objectId, {objectId}, {tbl_id: LC.RAW_TABLE, META_INFO});
+        getViewer(undefined, 'timeseries').showTable(req);
+    }
+}
+
+function checkForTimeSeriesData({tbl_id, highlightedRow}) {
+    const tableModel = getTblById(tbl_id) || {};
+    if (isTimeSeriesTable(tableModel)) {
+        return {show: true, title: 'View table as Time Series'};
+    } else {
+        const objectId = getCellValue(getTblById(tbl_id), highlightedRow, 'objectId');
+        return {show: Boolean(objectId), title: `View Time Series: ${objectId}`};
+    }
```

Comment:
if the table is DeepSource then you should just check for the existence of id. the Id in the deep source table is like the object id in the forced source table